client/lib/url/url-type.ts

```diff
@@ -0,0 +1,78 @@
+/**
+ * External dependencies
+ */
+import { URL as URLString } from 'types';
+import { Falsey } from 'utility-types';
+
+export enum URL_TYPE {
+	// A complete URL, with (at least) protocol and host.
+	// E.g. `http://example.com` or `http://example.com/path`
+	ABSOLUTE = 'ABSOLUTE',
+	// A URL with no protocol, but with a host.
+	// E.g. `//example.com` or `//example.com/path`
+	PROTOCOL_RELATIVE = 'PROTOCOL_RELATIVE',
+	// A URL with no protocol or host, but with a path starting at the root.
+	// E.g. `/` or `/path`
+	ROOT_RELATIVE = 'ROOT_RELATIVE',
```

Comment:
It might be good to follow the whatwg naming spec for these as much as possible. This might be `PATH_ABSOLUTE`.

https://url.spec.whatwg.org/#path-absolute-url-string

Reply:
Done, renamed to `PATH_ABSOLUTE`.